sage/api/src/error.rs

```diff
@@ -16,14 +21,23 @@ impl AsErrorCode for u8 {
 	}
 }
 
-impl AsErrorCode for Error {
+impl AsErrorCode for TransitionError {
+	fn as_error_code(&self) -> u8 {
+		match self {
+			TransitionError::InvalidTransitionId => 0,
+			TransitionError::TransferError => 1,
+			TransitionError::FeeError => 2,
+			TransitionError::Transition { error } => *error,
+		}
+	}
+}
+
+impl AsErrorCode for RuleError {
 	fn as_error_code(&self) -> u8 {
 		match self {
-			Error::InvalidTransitionId => 0,
-			Error::InvalidAssetLength => 1,
-			Error::TransferError => 2,
-			Error::FeeError => 3,
-			Error::Transition { error } => *error,
+			RuleError::AssetLength => 0,
+			RuleError::AssetOwnership => 1,
```

Comment:
Just to be extra unambiguous. I would recommend that the TransitionError and the RuleError don't have overlapping error types.